### PR TITLE
Adds get all articles API

### PR DESCRIPTION
**Pull request by cdesale**
Adds get all articles API --> task 4 - View a list of all articles

### CODE REVIEW

File: components/article-container.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { useState, useEffect } from "react";
+import { getAllArticles } from "../utils/api";
+
+export const ArticleContainer = () => {
+  const [articles, setArticles] = useState([]);
+
```
Comment (nc-verity-gregory) [inline components/article-container.jsx:6]: On a slow connection I have no idea if this is working as there is no loading state.

File: components/article-container.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  const [articles, setArticles] = useState([]);
+
+  useEffect(() => {
+    getAllArticles().then((data) => {
+      setArticles(data);
+    });
+  }, []);
+
+  return (
+    <div>
```
Comment (nc-verity-gregory) [inline components/article-container.jsx:14]: Could this be more semantic than a div?

File: components/article-container.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      {articles &&
+        articles.map((article) => {
+          return (
+            <div
+              style={{
```
Comment (nc-verity-gregory) [inline components/article-container.jsx:19]: Styling in line can bloat your files, you should absolutely put this in a css file.

File: components/article-container.jsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                <h3 style={{ margin: "0px" }}>{article.title}</h3>
+                <div
+                  style={{
+                    display: "flex",
+                    justifyContent: "space-between",
+                    margin: "0px",
+                    marginTop: "16px",
+                  }}
+                >
+                  <h5
```
Comment (nc-verity-gregory) [inline components/article-container.jsx:43]: h5?

File: src/App.jsx [hunk 5]
```diff
@@ -1,11 +1,13 @@
...
 import "./App.css";
 import { Header } from "../components/header";
+import { ArticleContainer } from "../components/article-container";
 
 function App() {
   return (
     <>
       <Header />
+      <ArticleContainer />
     </>
```
Comment (nc-verity-gregory) [inline src/App.jsx:11]: It is time to start implementing routes into this app

**Review by nc-verity-gregory — APPROVED** [review]
A good start on this, a little tight on a mobile layout there could be some larger margins.

Some notes and areas to address, but go ahead and merge.

File: components/article-container.jsx (hunk 1 above)
Comment (cdesale) [reply components/article-container.jsx:8]: Added a loading indicator.

File: components/article-container.jsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      setArticles(data);
+    });
+  }, []);
+
+  return (
+    <div>
+      {articles &&
+        articles.map((article) => {
+          return (
+            <div
```
[the remark this answers is not in the document]
Comment (cdesale) [reply components/article-container.jsx]: Moved the rendering into a separate component.

File: components/article-container.jsx (hunk 3 above)
Comment (cdesale) [reply components/article-container.jsx]: I'll keep this as is for now and update the styles in a later commit.

File: components/article-container.jsx (hunk 2 above)
Comment (cdesale) [reply components/article-container.jsx]: Yes, updated.

File: src/App.jsx (hunk 5 above)
Comment (cdesale) [reply src/App.jsx]: Updated to use Router.

**Pull request merged**